src/framework/auth/services/token/token.ts

```diff
@@ -16,11 +16,19 @@ export abstract class NbAuthToken {
 
 export interface NbAuthRefreshableToken {
   getRefreshToken(): string;
+  // Needed when refresh-token response does not repeat refresh_token value
```

Comment:
@alain-charles so it is possible that after refreshing the token server may not return a new refresh token? Just looked through the RFC and this part looks vague.

In any case, I would propose to move this functionality into the `refreshToken()` method of the strategy. In this case, we have the current token and can extract the refresh part right there with no need for all additional checks.

What do you think?

Reply:
I think that you are a amazing code architect :)